packages/block-editor/src/store/private-selectors.js

```diff
@@ -295,38 +295,70 @@ export const hasAllowedPatterns = createRegistrySelector( ( select ) =>
 	)
 );
 
+function mapUserPattern(
+	userPattern,
+	__experimentalUserPatternCategories = []
+) {
+	return {
+		name: `core/block/${ userPattern.id }`,
+		id: userPattern.id,
+		type: INSERTER_PATTERN_TYPES.user,
+		title: userPattern.title.raw,
+		categories: userPattern.wp_pattern_category.map( ( catId ) => {
+			const category = __experimentalUserPatternCategories.find(
+				( { id } ) => id === catId
+			);
+			return category ? category.slug : catId;
+		} ),
+		content: userPattern.content.raw,
+		syncStatus: userPattern.wp_pattern_sync_status,
+	};
+}
+
+export const getPatternBySlug = createRegistrySelector(
+	( select ) => ( state, patternName ) => {
+		// Only fetch reusable blocks if we know we need them.
+		// To do: maybe use the entity record API to retrieve the block by slug.
+		if ( patternName?.startsWith( 'core/block/' ) ) {
+			const _id = parseInt(
+				patternName.slice( 'core/block/'.length ),
+				10
+			);
+			const block = unlock( select( STORE_NAME ) )
+				.getReusableBlocks()
```

Comment:
Please remind me, why are all the selectors in this patch registry selectors, when they all select from the `block-editor` store? Why can't this `unlock( select() ).getReusableBlocks()` be a straighforward `getReusableBlocks`?

Reply:
Because it's a resolving selector :)